### PR TITLE
Convert Project to Multi Module

### PR DESCRIPTION
**Pull request by Shah-Parth**
* Converting project to multi module
* adding -SNAPSHOT to the release tag
* improving README

### CODE REVIEW

**Comment by Shah-Parth** [comment]
closes #10

File: gm-data-sdk/pom.xml [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                <enabled>true</enabled>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>never</updatePolicy>
+            </releases>
+        </repository>
+    </repositories>
+    <dependencies>
+        <!-- Commons Libraries -->
```
Comment (jeffusan) [inline gm-data-sdk/pom.xml]: My understanding is that provided scope dependencies will need to be explicitly added by programs using this library. Is this really what we want?

File: gm-data-sdk/pom.xml [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <source>${jre.version}</source>
+                    <target>${jre.version}</target>
+                    <encoding>UTF-8</encoding>
+                    <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                </configuration>
+            </plugin>
```
Comment (jeffusan) [inline gm-data-sdk/pom.xml]: Is this commented out because it's optional?

File: gm-data-sdk/pom.xml [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                        <executions>
+                            <execution>
+                                <id>sign-artifacts</id>
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>sign</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
```
Comment (jeffusan) [inline gm-data-sdk/pom.xml]: I recommend adding a comment/justification for commented out blocks. I prefer just removing them.

**Comment by jeffusan** [comment]
Are the p12 and truststore intended to be published with the OSS artifact?

**Comment by jeffusan** [comment]
The *ResponseHandlers seem to have duplicate logic, ie:
```
if status >= 300 {
 do something then throw exception
}
```
Can that be written once in the abstract class and removed from the subclasses?

**Comment by jeffusan** [comment]
`GreyMatterCloseableHttpClient` has a code block commented out:

```
    //TODO: uncomment once {@link GreyMatterDataStreamResponseHandler} is implemented
//    @Deprecated
//    public InputStream execute(final GreyMatterDataStreamRequest request) throws IOException, ClientProtocolException {
//        return execute(request, new GreyMatterDataStreamResponseHandler(), null);
//    }
```

`GreyMatterDataStreamResponseHandler` exists, but has a comment to the effect "this needs more thought".

What is the status of `GreyMatterDataStreamResponseHandler` and the intent going forward? Is it something that still needs to be implemented?

**Comment by Shah-Parth** [comment]
> `GreyMatterCloseableHttpClient` has a code block commented out:
> 
> ```
>     //TODO: uncomment once {@link GreyMatterDataStreamResponseHandler} is implemented
> //    @Deprecated
> //    public InputStream execute(final GreyMatterDataStreamRequest request) throws IOException, ClientProtocolException {
> //        return execute(request, new GreyMatterDataStreamResponseHandler(), null);
> //    }
> ```
> 
> `GreyMatterDataStreamResponseHandler` exists, but has a comment to the effect "this needs more thought".
> 
> What is the status of `GreyMatterDataStreamResponseHandler` and the intent going forward? Is it something that still needs to be implemented?

Yes, I need someone more familiar to take a look and tell me what I can do to provide users to leverage `GreyMatterDataStreamResponseHandler` instead of generic `HttpResponse`. If we can implement the logic inside `GreyMatterDataStreamResponseHandler` we can customize it to specific `gm-data` needs with standardize errors.

**Comment by Shah-Parth** [comment]
> The *ResponseHandlers seem to have duplicate logic, ie:
> 
> ```
>     @Override
>     public ArrayList<ResponseMetadata> handleResponse(final HttpResponse response)
>             throws IOException {
>         final StatusLine statusLine = response.getStatusLine();
>         final HttpEntity entity = response.getEntity();
>         if (statusLine.getStatusCode() >= 300) {
>             EntityUtils.consume(entity);
>             throw new HttpResponseException(statusLine.getStatusCode(),
>                     statusLine.getReasonPhrase());
>         }
>         return entity == null ? null : handleEntity(entity);
>     }
> 
>     /**
>      * Returns the entity as a body as a String.
>      */
>     @Override
>     public ArrayList<ResponseMetadata> handleEntity(final HttpEntity entity) throws IOException {
>         String entityString = EntityUtils.toString(entity);
>         Gson gson = new Gson();
> 
>         return gson.fromJson(entityString, new TypeToken<ArrayList<ResponseMetadata>>() {
>         }.getType());
>     }
> ```
> 
> Why is this duplicated in several classes instead of being just one class?

All of these classes are there as a placeholder for the initial starting point, we can for sure improve these classes and handle specific errors based on the particular endpoints. 
Yes, there is a lot of redundant code, but at the same time this is just an initial prototype of the SDK and it for sure needs more contribution. I don't think this is too critical at this point that it needs to be resolved in this PR.

File: gm-data-sdk/pom.xml (hunk 1 above)
Comment (Shah-Parth) [reply gm-data-sdk/pom.xml]: You're right... `compiled` is probably what we need to resolve the transitive dependencies.

**Comment by Shah-Parth** [comment]
Can someone please approve the PR so I can merge it.

**Pull request merged**